fixed bot.checkRole method

sergeme opened the pull request:
rolechecker.js: moved bot.checkRole from bot.js to constants/rolechecker.js. checking roles works now. returns only true if user is in at least on of the required roles. console output for logging purposes.

xp.js & reboot.js: bot.checkRole call fixed with await. granting xp works, console still throws some errors which have to do with level.js.

please add to master.

sergeme committed:
fixed bot.checkRole method

rolechecker.js: moved bot.checkRole from bot.js to constants/rolechecker.js. checking roles works now. returns only true if user is in at least on of the required roles. console output for logging purposes.

xp.js & reboot.js: bot.checkRole call fixed with await. granting xp works, console still throws some errors which have to do with level.js

milesflo commented on src/constants/rolechecker.js:
```diff
@@ -0,0 +1,27 @@
+const masterID = '127060142935113728'
+
+bot.checkRole = async (msg, roleArr) => {
+  var returnobj;
```
In JS, variables are almost always camel-case. So in the case of this one it should be
`var returnObj;`

milesflo replied:
And again here, this is going to be a boolean so call it something like `hasRole`

milesflo commented on src/constants/rolechecker.js:
```diff
@@ -0,0 +1,27 @@
+      if (await msg.guild.roles.find('name', roleArr[i]) != undefined) {
+        let foundRole = msg.guild.roles.find('name', roleArr[i]);
+        if (await msg.member.roles.has(foundRole.id)) {
+          authorroles.push(" "+foundRole.name)
```
Just push the name, not with the space

milesflo commented on src/constants/rolechecker.js:
```diff
@@ -0,0 +1,27 @@
+
+    if(authorroles.length > 0) { 
+      returnobj = true;
+      rolesStr = rolesStr + authorroles + " ";
```
Use the array.join method here...
`array.join(' ')`

sergeme committed:
implemented changes as requested

returnobj is now hasRole, authorroles is now authorRoles.
authorRoles gets converted to a string now with array.join(", ") for console output

removed bot.checkRole from bot.js pushed the wrong file to branch last time. added rolechecker.js as a constant to bot.js

bele321 commented:
implemented the changes, should be OK now. file bot.js has been updated with last push to this branch.

milesflo commented:
Great!
